# Statistical Analysis of High-Throughput Genomic and Transcriptomic Data
Fall/Herbst-semester 2021

## Lectures
Mondays 9.00-9.45 (Y27-H-46), 10.00-10.45 (Y27-H-46)

## Exercises
Monday 11.00-11.45 (Y27-H-46)

## Lecturers

[Dr. Hubert Rehrauer](http://www.fgcz.ch/the-center/people/rehrauer.html), Group Leader of Genome Informatics at [FGCZ](http://www.fgcz.ch/)  

[Prof. Dr. Mark Robinson](https://robinsonlabuzh.github.io/), Associate Professor of Statistical Genomics, [DMLS](https://www.mls.uzh.ch/en.html), UZH  

## Schedule

| Date  | Lecturer | Topic | Exercise | JC1 | JC2 |
| --- | --- | --- | --- | --- | --- |
| 20.09.2021  | Mark + Hubert  | admin; mol. bio. basics | R markdown; git(hub) | | |
| 27.09.2021  | Mark | interactive technology/statistics session  | group exercise: technology pull request | | |
| 04.10.2021  | Hubert | NGS intro; exploratory data analysis | EDA in R | | |
| 11.10.2021  | Hubert | mapping  | Rsubread | | |
| 18.10.2021  | Mark | limma + friends | linear model simulation + design matrices | | |
| 25.10.2021  | Hubert | RNA-seq quantification | RSEM  | X | X |
| 01.11.2021  | Mark | edgeR+friends 1 | basic edgeR/voom | [OUTRIDER: A Statistical Method for Detecting Aberrantly Expressed Genes in RNA Sequencing Data](https://www.cell.com/ajhg/fulltext/S0002-9297(18)30401-4) (BT, KN)  | [Powerful and robust non-parametric association testing for microbiome data via a zero-inflated quantile approach (ZINQ)](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01129-3) (RM, DS) |
| 08.11.2021  | Mark | edgeR+friends 2 | advanced edgeR/voom| [ZeitZeiger: supervised learning for high-dimensional data from an oscillatory system](https://dx.doi.org/10.1093%2Fnar%2Fgkw030) (TB, OF) |[SnapHiC:a computational pipeline to identify chromatin loops from single-cell Hi-C data](https://www.nature.com/articles/s41592-021-01231-2) (JS NH) |
| 15.11.2021  | TBA | hands-on session #1: RNA-seq  | FASTQC/Salmon/etc. | [Differential abundance testing on single-cell data using k-nearest neighbor graphs](https://www.nature.com/articles/s41587-021-01033-z) (VW, JL) | X |
| 22.11.2021  | Hubert | single-cell 1: preprocessing, dim. reduction, clustering | clustering | [title](link) (HJ,SK) | [A variable selection approach for highly correlated predictors in high-dimensional genomic data](https://academic.oup.com/bioinformatics/article/37/16/2238/6146520?login=true) (SV,PN) |
| 29.11.2021  | TBA | hands-on session #2: cytometry  | cytof null comparison | X | X |
| 06.12.2021  | Mark | single-cell 2: clustering, marker gene DE  | marker gene DE | [Causal inference in genetic trio studies](https://www.pnas.org/content/117/39/24117) (CG, DC) | X |
| 13.12.2021  | Pierre-Luc | hands-on session #3: single-cell RNA-seq (cell type definition, differential state)  | full scRNA-seq pipeline | X | X |
| 20.12.2021  | Mark | loose ends: HMM, EM, robustness   | segmentation, peak finding | X | X |

 
## Course material

Assuming you have git installed locally, you can check out the entire set of course materials with the following command (from command line):
```
git clone https://github.com/sta426hs2020/material.git
```  
and get updates by running `git pull` at any later time in the same directory.

Alternatively, for a ZIP file of the repository, you can click on the (green) 'Code' button (top right of main panel) and then click 'Download ZIP'.

